src/main.rs

```diff
@@ -36,26 +33,28 @@ struct Cli {
 
 fn main() {
     let args = Cli::from_args();
-    let mut last_time = 0.0;
-    let refresh_rate = time::Duration::from_nanos((args.refresh_rate * ONE_BILLION) as u64);
+    let refresh_rate = time::Duration::from_secs_f32(args.refresh_rate);
     let mut ui = UI::new(
         &args.command,
         args.target_result,
         args.history_len,
         args.show_regression_line.or(Some(true)).unwrap(),
-        args.show_target_line.or(Some(false)).unwrap()
+        args.show_target_line.or(Some(false)).unwrap(),
     );
     let mut keys = async_stdin().keys();
+    // Substract refresh_rate to display the first frame immediatly
+    let mut last_time = Instant::now() - refresh_rate;
 
-    loop{
-        let current_time = SystemTime::now().duration_since(UNIX_EPOCH).unwrap().as_secs_f64();
-        if refresh_rate.as_secs_f64() < current_time - last_time {
+    loop {
+        let current_time = Instant::now();
+        if refresh_rate < current_time - last_time {
             ui.evaluate();
             last_time = current_time;
         }
         if !ui.event_handler(keys.next()) {
-            ui.clean_up_terminal();
-            break
+            break;
         }
-    }  
-}
\ No newline at end of file
+    }
+
+    ui.clean_up_terminal();
```

Comment:
I moved the terminal clean up outside the loop to ensure this is called at the end of the main loop and not only under some arbitrary conditions.